File: frontend/apprestaurant/src/app/core/services/configuracion.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ConfiguracionRequest, ConfiguracionResponse } from '../models/configuracion.models';
import { ApiService } from './api.service';

@Injectable({ providedIn: 'root' })
export class ConfiguracionService {
  private http = inject(HttpClient);
  private api = inject(ApiService);

  private url = `${this.api.baseUrl}/configuracion`;

  obtenerTodas(): Observable<ConfiguracionResponse[]> {
    return this.http.get<ConfiguracionResponse[]>(`${this.url}`);
  }

  obtenerPorClave(clave: string): Observable<ConfiguracionResponse> {
    return this.http.get<ConfiguracionResponse>(`${this.url}/${clave}`);
  }

  crear(data: ConfiguracionRequest): Observable<ConfiguracionResponse> {
    return this.http.post<ConfiguracionResponse>(`${this.url}`, data);
  }

  actualizar(id: number, data: ConfiguracionRequest): Observable<ConfiguracionResponse> {
    return this.http.put<ConfiguracionResponse>(`${this.url}/${id}`, data);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/${id}`);
  }
}

File: src/main/java/pe/edu/upeu/sys_api_restaurant/service/ConfiguracionService.java
package pe.edu.upeu.sys_api_restaurant.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import pe.edu.upeu.sys_api_restaurant.exception.ResourceNotFoundException;
import pe.edu.upeu.sys_api_restaurant.dto.*;
import pe.edu.upeu.sys_api_restaurant.entity.Configuracion;
import pe.edu.upeu.sys_api_restaurant.mapper.ConfiguracionMapper;
import pe.edu.upeu.sys_api_restaurant.repository.ConfiguracionRepository;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ConfiguracionService {

    private final ConfiguracionRepository repository;

    @Transactional(readOnly = true)
    public List<ConfiguracionResponse> obtenerTodas() {
        return repository.findAll()
                .stream()
                .map(ConfiguracionMapper::toConfiguracionResponse)
                .toList();
    }

    @Transactional(readOnly = true)
    public ConfiguracionResponse obtenerPorClave(String clave) {
        return repository.findByClave(clave)
                .map(ConfiguracionMapper::toConfiguracionResponse)
                .orElseThrow(() -> new ResourceNotFoundException("Configuracion no encontrada con clave: " + clave));
    }

    @Transactional
    public ConfiguracionResponse crear(ConfiguracionRequest request) {
        Configuracion entity = ConfiguracionMapper.toConfiguracion(request);
        return ConfiguracionMapper.toConfiguracionResponse(repository.save(entity));
    }

    @Transactional
    public ConfiguracionResponse actualizar(Integer id, ConfiguracionRequest request) {
        Configuracion existing = repository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Configuracion no encontrada con id: " + id));
        existing.setClave(request.clave());
        existing.setValor(request.valor());
        existing.setDescripcion(request.descripcion());
        return ConfiguracionMapper.toConfiguracionResponse(repository.save(existing));
    }

    @Transactional
    public void eliminar(Integer id) {
        Configuracion existing = repository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Configuracion no encontrada con id: " + id));
        repository.delete(existing);
    }

    @Transactional(readOnly = true)
    public String obtenerValor(String clave) {
        return repository.findByClave(clave)
                .map(Configuracion::getValor)
                .orElseThrow(() -> new ResourceNotFoundException("Configuracion no encontrada con clave: " + clave));
    }
}
